docs/guides/js-waku/store-retrieve-messages.md

```diff
@@ -0,0 +1,219 @@
+---
+title: Retrieve Messages Using Store
+---
+
+This guide provides detailed steps to create a light node for retrieving and filtering historical messages using the [Store protocol](/overview/concepts/protocols#store).
+
+## Create a Light Node
+
+Use the `createLightNode()` function to create a light node and interact with the Waku Network:
+
+```js
+import { createLightNode } from "@waku/sdk";
+
+// Create and start a light node
+const node = await createLightNode({ defaultBootstrap: true });
+await node.start();
+```
+
+## Connect to Store Peers
+
+Use the `waitForRemotePeer()` function to wait for the node to connect with store peers:
+
+```js
+import { waitForRemotePeer, Protocols } from "@waku/sdk";
+
+// Wait for a successful peer connection
+await waitForRemotePeer(node, [Protocols.Store]);
+```
+
+## Choose a Content Topic
+
+[Choose a content topic](/overview/concepts/content-topics) for filtering the messages to retrieve and create a `decoder` for [message decryption](https://rfc.vac.dev/spec/26/):
```

Comment:
A decoder does not decrypt messages per se.

Waku Messages are protobuf encoded over the wire.

a decoder decodes the waku messages from protobuf format, as per the wire format spec: https://rfc.vac.dev/spec/14/#wire-format

If a decoder from `@waku/message-encryption` is used, then the decoder decodes from protobuf AND decrypt the message.